### PR TITLE
Temp appmesh translator

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/supergloo/issues/231

### CODE REVIEW

File: pkg/config/appmesh/syncer_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package appmesh
```
Comment (ilackarms) [inline pkg/config/appmesh/syncer_test.go]: rm

File: pkg/translator/appmesh/config_test.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+	gloov1 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	"github.com/solo-io/supergloo/cli/pkg/helpers/clients"
+	"github.com/solo-io/supergloo/pkg/api/custom/clients/kubernetes"
+	kubecustom "github.com/solo-io/supergloo/pkg/api/external/kubernetes/core/v1"
+	v1 "github.com/solo-io/supergloo/pkg/api/v1"
+	"github.com/solo-io/supergloo/pkg/webhook/test"
+)
+
+var appMesh = func(name string) *v1.Mesh {
```
Comment (ilackarms) [inline pkg/translator/appmesh/config_test.go]: put this in inputs `inputs.InputAppMeshMesh`

File: pkg/translator/appmesh/initialize.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+				// Env value is expected to have the following format
+				// - name: "APPMESH_VIRTUAL_NODE_NAME"
+				//   value: "mesh/meshName/virtualNode/virtualNodeName"
+				vnNameParts := strings.Split(env.Value, "/")
+				if len(vnNameParts) != 4 {
+					return nil, errors.Errorf("unexpected format for %s env for pod %s.%s. Expected format is [%s] but found [%s]",
+						PodVirtualNodeEnvName, kubePod.Namespace, kubePod.Name, "mesh/meshName/virtualNode/virtualNodeName", env.Value)
+				}
+
+				if meshName := vnNameParts[1]; meshName == mesh.Metadata.Name {
```
Comment (ilackarms) [inline pkg/translator/appmesh/initialize.go]: open an issue to do a follow up pr to better correlate mesh crd namespace+name with appmesh mesh name

File: pkg/translator/appmesh/initialize.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+			}
+		}
+	}
+
+	if info != nil {
+		for _, initContainer := range kubePod.Spec.InitContainers {
+			for _, env := range initContainer.Env {
+				if env.Name == PodPortsEnvName {
+					for _, portStr := range strings.Split(env.Value, ",") {
+						ui64, err := strconv.ParseUint(strings.Trim(portStr, " "), 10, 32)
```
Comment (ilackarms) [inline pkg/translator/appmesh/initialize.go]: nit: `strings.TrimSpaces`

File: pkg/translator/appmesh/initialize.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+			podInfo[pod].upstreams = append(podInfo[pod].upstreams, us)
+		}
+	}
+
+	var appMeshUpstreamList gloov1.UpstreamList
+	for us := range appMeshUpstreamInfo {
+		appMeshUpstreamList = append(appMeshUpstreamList, us)
+	}
+
+	return appMeshUpstreamInfo, appMeshUpstreamList, nil
```
Comment (ilackarms) [inline pkg/translator/appmesh/initialize.go]: sort for idempotency appMeshUpstreamList

File: pkg/translator/appmesh/configuration.go [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+		upstreamInfo:     appMeshUpstreamInfo,
+		upstreamList:     appMeshUpstreamList,
+		VirtualNodeLabel: awsMesh.VirtualNodeLabel,
+		MeshName:         mesh.Metadata.Name,
+		VirtualServices:  make(PodVirtualService),
+		VirtualNodes:     make(PodVirtualNode),
+	}, nil
+}
+
+func (c *awsAppMeshConfiguration) initialize() error {
```
Comment (ilackarms) [inline pkg/translator/appmesh/configuration.go]: make this part of the constructor, remove

File: pkg/translator/appmesh/initialize.go [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+			}
+		}
+	}
+	return nil, errors.Errorf("unable to find upstream with selector %s", vnLabel)
+}
+
+func podAppmeshConfig(info *podInfo, meshName, hostName string) (*appmesh.VirtualNodeData, *appmesh.VirtualServiceData) {
+	var vn *appmesh.VirtualNodeData
+	var vs *appmesh.VirtualServiceData
+	listeners := make([]*appmesh.Listener, len(info.ports))
```
Comment (ilackarms) [inline pkg/translator/appmesh/initialize.go]: remove these vars

File: pkg/translator/appmesh/initialize.go [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+				Protocol: &protocol,
+				Port:     &port,
+			},
+		}
+	}
+	vn = &appmesh.VirtualNodeData{
+		MeshName:        &meshName,
+		VirtualNodeName: &info.virtualNodeName,
+		Spec: &appmesh.VirtualNodeSpec{
+			Backends:  []*appmesh.Backend{},
```
Comment (ilackarms) [inline pkg/translator/appmesh/initialize.go]: comment that backends gets filled later

File: pkg/translator/appmesh/initialize.go (hunk 3 above)
Comment (EItanya) [reply pkg/translator/appmesh/initialize.go]: done

**Pull request merged**